### DISCUSSION TITLE
Very slow at generating CSS from Svelte or HTML files

### DISCUSSION THREAD

**Discussion by matthewmueller** [tailwindlabs/tailwindcss#5816] [Help] [answered]
**What version of Tailwind CSS are you using?**

v2.2.16

**What build tool (or framework if it abstracts the build tool) are you using?**

None

**What version of Node.js are you using?**

v14.16.0

**What browser are you using?**

N/A

**What operating system are you using?**

OSX

**Describe your issue**

I'm quite new to Tailwind, but was surprised by how slow extracting class names and generating CSS from Svelte was.

Given the following `Header.svelte` file:

```svelte
<div class="bg-orange-400 p-4 flex items-center">
  <ul class="flex items-center">
    <li><img src="/y18.gif" alt="" /></li>
    <li class="pl-3 font-mono text-orange-900 text-lg">Hacker News</li>
  </ul>

  <ul class="ml-auto flex items-center">
    <li class="text-sm font-semibold text-white">New</li>
    <li class="pl-6 text-sm font-semibold text-white">Ask</li>
    <li class="pl-6 text-sm font-semibold text-white">Show</li>
  </ul>
</div>

<style>
  ul {
    list-style: none;
  }
</style>
```

And `tailwind.config.js`:

```js
module.exports = {
  jit: true,
  corePlugins: {
    preflight: false,
  },
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
```

I'd like to extract and generate CSS from that Svelte file. I ran the following command:

```sh
$ npx tailwindcss --purge view/Header.svelte
# Printed out correct CSS
Done in 3287ms.
```

It took over 3 seconds to generate CSS. Intuitively, I would have assumed that it parsed the HTML, looked for class names, then keyed those class names with the CSS attributes. Is it doing a lot more?

I might also be using the command wrong, maybe I'm using the production command? I also tried with `--input`, but that seemed to expect CSS files, not Svelte files.

Thanks for your help!

**Comment by matthewmueller** [comment]
**Update:** I also ran it with .html, the performance is around the same:

```sh
$ npx tailwindcss --purge view/Header.html
# CSS
Done in 3292ms.
```

**Comment by adamwathan** [comment, marked as answer]
Hey! Use `mode: ‘jit’` not `jit: true` to actually enable JIT mode, should be fast then.

**Reply by matthewmueller** [reply]
```sh
# Changed to `mode: "jit"` in tailwind.config.js
$ npx tailwindcss --purge view/Header.svelte
# CSS...
Done in 94ms.
```

:facepalm:   thanks!
